repository-data/webfiles/src/main/resources/site/freemarker/common/api-catalogue.ftl

```diff
@@ -49,23 +50,38 @@
 
         <#if alphabetical_hash??>
             <div class="grid-row">
-                <div class="column column--one-third page-block page-block--sidebar">
+                <div class="column column--one-third page-block page-block--sidebar sticky sticky--top">
                     <@alphabeticalFilterNav alphabetical_hash></@alphabeticalFilterNav>
 
                     <#if filtersModel?? && !filtersModel.isEmpty()>
                         <div class="article-section-nav-wrapper">
-                            <div class="article-section-nav">
-                                <h2 class="article-section-nav__title">Filters</h2>
-                                <nav class="filters">
+                            <div class="article-section-nav filters">
+
+                                <h2 class="article-section-nav__title">
+                                    <span class="filter-head__title">Filters</span>
+                                    <@hst.renderURL var="resetUrl"/>
+                                    <a class="filter-head__reset button button--tiny" href="${resetUrl}" title="Reset">Reset</a>
+                                </h2>
+
+                                <nav>
                                     <ul>
                                         <#list filtersModel.sections as section>
                                             <#if section.displayed>
                                             <li>
-                                                <input id="toggler_${section.key}" type="checkbox" <#if section.expanded>checked</#if>/>
-                                                <label for="toggler_${section.key}" class="section-heading <#if section.expanded>selected</#if>">${section.displayName}</label>
+                                                <input id="toggler_${section.key}" type="checkbox" <#if section.expanded>checked</#if> aria-hidden="true"/>
+                                                <#if section.expanded>
+                                                    <#assign "expansionArrowClass" = "expansion-arrow selected" />
```

Comment:
does the following work in place of this if/else block?
```
<#assign "expansionArrowClass" = "expansion-arrow <#if section.expanded>selected</#if>" />
```

Reply:
Indeed! Your version is much cleaner; thanks, Harry :+1: 

I'm changing it now.

Reply:
As it turns out, strings can't be interpolated with the `<#if...>` constructs like this.

However, Freemarker contains a useful '[built-in|(https://freemarker.apache.org/docs/dgui_template_valueinsertion.html#autoid_17)', which, combined with string contactenation, works as a good equivalent of Java's ternary operator. The following resulting line seems to works:

`<#assign "expansionArrowClass" = "expansion-arrow " + section.expanded?string("selected", "") />`

I'm glad you gave me an excuse to look into it a bit more :wink: